File: frontend/src/utils/verification.ts
/** 资源校验证据 — 与后端 ResourceVerificationResult 对齐 */

export type VerifyStatus = 'passed' | 'warning' | 'failed'

export interface GroundedChunkRef {
  id: string
  title?: string
  snippet?: string
}

export interface ResourceVerification {
  resource_id?: number
  resource_type?: string
  course_id?: string
  chapter_id?: string
  verifier_status?: VerifyStatus
  safety_status?: VerifyStatus
  grounded_chunks?: GroundedChunkRef[]
  hallucination_risks?: string[]
  unsupported_claims?: string[]
  sensitive_risks?: string[]
  prompt_injection_risks?: string[]
  retry_count?: number
  skip_reason?: string
  final_decision?: 'publish' | 'draft' | 'blocked'
  risk_label?: string
  evidence_count?: number
  created_at?: string
}

export function getResourceVerification(meta: Record<string, unknown> | undefined): ResourceVerification | null {
  if (!meta) return null
  const v = meta.verification
  if (v && typeof v === 'object') return v as ResourceVerification
  const panel = meta.safety_panel as Record<string, unknown> | undefined
  if (panel?.verification && typeof panel.verification === 'object') {
    return panel.verification as ResourceVerification
  }
  return null
}

export function verificationDisplayTag(meta: Record<string, unknown>): {
  label: string
  type: 'success' | 'warning' | 'danger' | 'info'
  riskLabel: string
} {
  const v = getResourceVerification(meta)
  if (v?.risk_label) {
    const risk = v.risk_label
    if (risk === '无风险' || v.final_decision === 'publish') {
      return { label: '内容校验通过', type: 'success', riskLabel: risk }
    }
    if (v.final_decision === 'blocked' || v.safety_status === 'failed') {
      return { label: '安全审查未通过', type: 'danger', riskLabel: risk }
    }
    return { label: '内容校验告警', type: 'warning', riskLabel: risk }
  }
  if (meta?.verified === true || meta?.status === 'published') {
    return { label: '已校验', type: 'success', riskLabel: '无风险' }
  }
  if (meta?.status === 'draft') {
    return { label: '待校验', type: 'warning', riskLabel: '待复核' }
  }
  return { label: '未校验', type: 'info', riskLabel: '未校验' }
}

File: backend/schemas/verification.py
"""资源防幻觉与内容安全校验结果。"""

from __future__ import annotations

from datetime import datetime, timezone
from typing import Literal

from pydantic import BaseModel, Field

VerifyStatus = Literal["passed", "warning", "failed"]
FinalDecision = Literal["publish", "draft", "blocked"]


class GroundedChunkRef(BaseModel):
    id: str
    title: str = ""
    snippet: str = ""
    module_id: str = ""
    chapter_title: str = ""
    section_title: str = ""
    source_path: str = ""
    relevance_score: float = 0.0


class ResourceVerificationResult(BaseModel):
    resource_id: int = 0
    resource_type: str = ""
    course_id: str = "data_structures_algorithms"
    chapter_id: str = ""
    verifier_status: VerifyStatus = "warning"
    safety_status: VerifyStatus = "warning"
    grounded_chunks: list[GroundedChunkRef] = Field(default_factory=list)
    hallucination_risks: list[str] = Field(default_factory=list)
    unsupported_claims: list[str] = Field(default_factory=list)
    sensitive_risks: list[str] = Field(default_factory=list)
    prompt_injection_risks: list[str] = Field(default_factory=list)
    retry_count: int = 0
    skip_reason: str = Field(default="", description="跳过文本校验时的原因")
    final_decision: FinalDecision = "draft"
    risk_label: str = Field(default="未校验", description="前端展示：无风险/可能幻觉/安全警告/已重试")
    evidence_count: int = 0
    created_at: str = Field(default_factory=lambda: datetime.now(timezone.utc).isoformat())

    def to_meta_dict(self) -> dict:
        return self.model_dump()
